syft/generic/frameworks/hook/hook.py

```diff
@@ -145,6 +145,23 @@ def overloaded_native_method(self, *args, **kwargs):
             Operate the hooking
             """
 
+            def match_natural2fixprec(t1, t2):
```

Comment:
Q: Could you move this from here - ideally the generic/frameworks should have no knowledge about the "more" specific tensors we implement (like FPT or AST)